pkg/cloudprovider/azure.go

```diff
@@ -437,9 +449,9 @@ func (a *Azure) getNetworkInterfaces(instance *compute.VirtualMachine) ([]networ
 		}
 	}
 	// Get the rest and append that.
-	for _, netif := range *instance.NetworkProfile.NetworkInterfaces {
-		if netif.NetworkInterfaceReferenceProperties != nil && ((netif.Primary != nil && !*netif.Primary) || netif.Primary == nil) {
-			intf, err := a.getNetworkInterface(*netif.ID)
+	for _, netif := range instance.Properties.NetworkProfile.NetworkInterfaces {
+		if netif.Properties != nil && ((netif.Properties.Primary != nil && !ptr.Deref(netif.Properties.Primary, false)) || netif.Properties.Primary == nil) {
```

Comment:
I can this can be simplified to
```suggestion
		if netif.Properties != nil && !ptr.Deref(netif.Properties.Primary, false) {
```